test/e2e/slow_tests.py

```diff
@@ -0,0 +1,255 @@
+
+        backend = self.real_device_backend
+
+        def _job_callback(job_id, job_status, cjob, **kwargs):
+            self.simple_job_callback(job_id, job_status, cjob, **kwargs)
+            if job_status is JobStatus.RUNNING:
+                job_properties[0] = cjob.properties()
+                cancel_job(cjob)
+
+        job_properties = [None]
+        large_qx = get_large_circuit(backend=backend)
+        job = backend.run(transpile(large_qx, backend=backend))
+        job.wait_for_final_state(wait=None, callback=_job_callback)
+        self.assertIsNotNone(job_properties[0])
+
+    def test_error_message_device(self):
+        """Test retrieving job error messages from a device backend."""
+        backend = self.real_device_backend
+        job = submit_job_one_bad_instr(backend)
+        job.wait_for_final_state(wait=300, callback=self.simple_job_callback)
+
+        rjob = self.dependencies.provider.backend.job(job.job_id())
+
+        for q_job, partial in [(job, False), (rjob, True)]:
+            with self.subTest(partial=partial):
+                with self.assertRaises(IBMJobFailureError) as err_cm:
+                    q_job.result(partial=partial)
+                for msg in (err_cm.exception.message, q_job.error_message()):
+                    self.assertIn("bad_instruction", msg)
+                    self.assertIsNotNone(
+                        re.search(r"Error code: [0-9]{4}\.$", msg), msg
+                    )
+
+    def test_headers_in_result_devices(self):
+        """Test that the qobj headers are passed onto the results for devices."""
+        custom_header = {"x": 1, "y": [1, 2, 3], "z": {"a": 4}}
+
+        # TODO Use circuit metadata for individual header when terra PR-5270 is released.
+        # qobj.experiments[0].header.some_field = 'extra info'
+
+        quantum_register = QuantumRegister(1)
+        classical_register = ClassicalRegister(1)
+
+        qc1 = QuantumCircuit(quantum_register, classical_register, name="circuit0")
+        qc1.h(quantum_register[0])
+        qc1.measure(quantum_register, classical_register)
+        job = self.real_device_backend.run(
+            transpile(qc1, backend=self.real_device_backend), header=custom_header
+        )
+        job.wait_for_final_state(wait=300, callback=self.simple_job_callback)
+        result = job.result()
+        self.assertTrue(custom_header.items() <= job.header().items())
+        self.assertTrue(custom_header.items() <= result.header.to_dict().items())
+        # self.assertEqual(result.results[0].header.some_field, 'extra info')
+
+    def test_websockets_device(self):
+        """Test checking status of a job via websockets for a device."""
+        job = self.real_device_backend.run(
+            transpile(ReferenceCircuits.bell(), self.real_device_backend), shots=1
+        )
+
+        # Manually disable the non-websocket polling.
+        # job._api_client._job_final_status_polling = self._job_final_status_polling
```

Comment:
Not sure what the exact purpose of line 251 is but `_job_final_status_polling` causes the test to fail